### DISCUSSION TITLE
p-fileupload in production cors error

### DISCUSSION THREAD

**Discussion by DimRX** [primefaces/community#1864] [PrimeNG]
Hello,

Having put my site into production, I'm noticing that my p-fileUpload is giving me a CORS error when I want to upload files that I'm going to store in a mongo BD on my backend.

Here's my code : 
 
```
 <p-fileUpload id="upload-images-profesional" name="demo[]" url="https://www.primefaces.org/cdn/api/upload.php"
                                          [fileLimit]=this.IMAGES_LIMIT
                                          (onUpload)="onUpload($event)"
                                          [multiple]="true"
                                          accept="image/*"
                                          [maxFileSize]="5000000">
```

And here's my error message : 

`has been blocked by CORS policy: Request header field authorization is not allowed by Access-Control-Allow-Headers in preflight response.`

**Comment by Yazan-Abuawwad** [comment]
You must put the link in the whitelist.
url="https://www.primefaces.org/cdn/api/upload.php"

**Comment by qwadrox** [comment]
> https://www.primefaces.org/cdn/api/upload.php

Hi @DimRX 

It would be best not to use that example URL from the primefaces cdn. It's just a script that returns {"success": true}
https://primeng.org/fileupload#api.fileupload.props.url
You should provide your server URL in that property. 

And yes, as @Yazan-Abuawwad mentioned you probably have a Cors issue that should be resolved on the server side. 
Luckily there is a Youtube video you can follow along which is showcasing your problem and the property usage of the URL property: 
https://youtu.be/F-T3v_FyCD0?t=639

In the video, he has an ASP.net backend but the logic is the same in every framework. 
Search Enable CORS "Your backend framework".
